src/treetensornetwork/ttno.jl

```diff
@@ -0,0 +1,148 @@
+"""
+    TreeTensorNetworkOperator <: AbstractITensorNetwork
+
+A finite size tree tensor network operator type. 
+Keeps track of the orthogonality center.
+
+# Fields
+
+- itensor_network::ITensorNetwork
+- ortho_lims::Vector{Tuple}: A vector of vertices defining the orthogonality limits.
+"""
+mutable struct TreeTensorNetworkOperator <: AbstractTreeTensorNetwork
```

Comment:
I wonder if we can get away with just having a `TreeTensorNetwork` type that is treated as an operator or a state depending on the context.

If we really need to distinguish them for the sake of dispatch, we could have a wrapper type like `Map(ttn)` or `Operator(ttn)`, which could be like an `ITensorNetworkMap`: https://github.com/ITensor/ITensors.jl/blob/v0.3.20/src/ITensorNetworkMaps/ITensorNetworkMaps.jl

Reply:
Tracking in #26.